volumes.c

```diff
@@ -185,11 +186,28 @@ struct pv_disk* pv_disk_add(struct pv_state *s)
 static int pv_volume_mount_handler(struct pv_volume *v, char *action)
 {
 	struct pv_disk *d = v->disk;
+	char path[PATH_MAX];
+	char *disk_path = strdup(d->path);
+	char *disk_name = NULL;
 	char *command = NULL;
 	char *crypt_type;
 	int ret;
 	int wstatus;
 
+	disk_name = strtok(basename(disk_path), ",");
+	if (disk_name == NULL) {
+		free(disk_path);
+		return -EINVAL;
+	}
+
+	pv_paths_storage_crypt_file(path, PATH_MAX, disk_name);
```

Comment:
isnt this rather a pv_paths_storage_mounted_disk_path? that gets input "handler_type" and disk_name and then is constructed like /media/pv/$handler_type/$disk_name ?

Reply:
Not sure if the current changes are fine. Please revisit @asac